src/attr/_make.py

```diff
@@ -484,6 +487,9 @@ def _transform_attrs(cls, these, auto_attribs, kw_only, collect_by_mro):
         for attr_name, ca in ca_list
     ]
 
+    if transform_hook is not None:
+        own_attrs = [transform_hook(cls, a) for a in own_attrs]
```

Comment:
Out of curiosity: why did you choose to call them one by one and not all at once?

I have a gut feeling that an `own_atts = transform_hooks(cls, own_attrs)` wouldn't be just faster (less method calls) but also allow for more use-cases where you need see all attributes? 🤔

Reply:
This is a good point. When I started work on this feature, I called the hook in another place where I *needed* to call it once for each attrib. This is now no longer necessary and my default auto-convert-hook would actually benefit from making this one-call-for-all. :+1:

Reply:
What was the use-case for needing to call them once? Can't you simulate it using an intermediate function?

Reply:
There never was a use case. It was just the natural way of doing it were I originally implemented it. I think there is no drawback in making it `hook(cls: type, fields: List[Attribute]) -> List[Attribute])`

Reply:
Wonderful, pls do. Let’s provide the users with as much context as necessary.